File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/get/MultiGetResponse.java
package fr.myprysm.vertx.elasticsearch.action.get;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@DataObject(generateConverter = true)
public class MultiGetResponse {

    private List<GetResponse> responses;
    private List<GetFailure> failures;


    public MultiGetResponse(JsonObject json) {
        MultiGetResponseConverter.fromJson(json, this);
    }

    public JsonObject toJson() {
        JsonObject json = new JsonObject();
        MultiGetResponseConverter.toJson(this, json);
        return json;
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/get/MultiGetResponse.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.get

import fr.myprysm.vertx.elasticsearch.action.get.MultiGetResponse
import fr.myprysm.vertx.elasticsearch.action.get.GetFailure
import fr.myprysm.vertx.elasticsearch.action.get.GetResponse

fun MultiGetResponse(
        failures: Iterable<fr.myprysm.vertx.elasticsearch.action.get.GetFailure>? = null,
        responses: Iterable<fr.myprysm.vertx.elasticsearch.action.get.GetResponse>? = null): MultiGetResponse = fr.myprysm.vertx.elasticsearch.action.get.MultiGetResponse().apply {

    if (failures != null) {
        this.setFailures(failures.toList())
    }
    if (responses != null) {
        this.setResponses(responses.toList())
    }
}

